### PR TITLE
test: use `T.Setenv` to set env vars in tests

### PR DESCRIPTION
**Pull request by Juneezee**
**- What I did**
A testing cleanup.

This PR replaces `os.Setenv` with `t.Setenv`. Starting from Go 1.17, we can use `t.Setenv` to set environment variable in test. The environment variable is automatically restored to its original value when the test and all its subtests complete.

Reference: https://pkg.go.dev/testing#T.Setenv

**- How I did it**

Search for usages of `os.Setenv` in `*_test.go` using GoLand, and replace them with `t.Setenv` accordingly.

**- How to verify it**

All tests should continue to pass, except those that are already failing before this PR.

**- Description for the changelog**
```
test: use `T.SetEnv` to set env vars in tests
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by tianon — APPROVED** [review]
This seems fine to me too -- it will potentially cause issues with some packagers who will have to backport Go 1.17 if they want to backport newer Docker releases, but I don't think there's much more we can do (reasonably) to help there.

**Comment by thaJeztah** [comment]
Thanks @Juneezee ! 

As Tianon mentioned in the previous comment; we discussed this change in our maintainers call, and agreed that it's ok to make these changes (and it's nice to use more of Go's built-in functionality).


As a follow-up we should have a look at existing uses of `gotest.tools` `env.Patch()`, which can now be replaced by `t.SetEnv()` (and has been deprecated recently in favour of that); https://github.com/gotestyourself/gotest.tools/blob/9c6ef764e11bed6bc0d9455a775d8cac665e1d1d/env/env.go#L18-L26

(We'd still need to keep / want `gotest.tools` `env.PatchAll()`)

Let me know if you're interested to work on that!

**Comment by thaJeztah** [comment]
oh! I just realised I was touching much of those in https://github.com/moby/moby/pull/43340, so perhaps I should just include that change in that one 😅

**Pull request merged**
